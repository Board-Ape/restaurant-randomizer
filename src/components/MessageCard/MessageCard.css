.message-displa {
  color: white;
  font-size: 40px; }
