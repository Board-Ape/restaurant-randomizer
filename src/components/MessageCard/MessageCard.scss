.message-display {
  color: white;
  font-size: 40px;
}

div {
  color: white;
}
