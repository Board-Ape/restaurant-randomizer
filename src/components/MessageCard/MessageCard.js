import React from 'react';
import './MessageCard.css';
import PropTypes from 'prop-types';

const MessageCard = (props) => {
  return (
    <div className='message-container'>
      <div className='message-display'>{props.eventName}</div>
    </div>
  );
};

export default MessageCard;

MessageCard.propTypes = {
  eventName: PropTypes.array
};
