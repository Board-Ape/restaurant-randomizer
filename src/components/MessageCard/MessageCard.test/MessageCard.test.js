import React from 'react';
import MessageCard from '../MessageCard';
import { shallow } from 'enzyme';

const mockData = {another: 'Day', some: 'Other'};

describe('MessageCard', () => {
  let renderedMessageCard;

  beforeEach(() => {
    renderedMessageCard = shallow(
      <MessageCard
        data={mockData}
        eventName={['hello', 'there']}
      />
    );
  });

  it('should exist', () => {
    expect(renderedMessageCard).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(renderedMessageCard).toMatchSnapshot();
  });

  it('should render new message cards', () => {
    const expected = [mockData].length;

    expect(renderedMessageCard.find('.message-display').length).toEqual(expected);
  });
});
