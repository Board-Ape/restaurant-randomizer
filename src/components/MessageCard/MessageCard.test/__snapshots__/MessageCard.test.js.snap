// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MessageCard
    data={
      Object {
        "another": "Day",
        "some": "Other",
      }
    }
    eventName={
      Array [
        "hello",
        "there",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="message-display"
      >
        hello
        there
      </div>,
      "className": "message-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "hello",
          "there",
        ],
        "className": "message-display",
      },
      "ref": null,
      "rendered": Array [
        "hello",
        "there",
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="message-display"
        >
          hello
          there
        </div>,
        "className": "message-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "hello",
            "there",
          ],
          "className": "message-display",
        },
        "ref": null,
        "rendered": Array [
          "hello",
          "there",
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
